src/FakeItEasy/A.cs

```diff
@@ -44,7 +44,7 @@ public static T Fake<T>(Action<IFakeOptions<T>> optionsBuilder)
         {
             Guard.AgainstNull(optionsBuilder, nameof(optionsBuilder));
 
-            return (T)FakeAndDummyManager.CreateFake(typeof(T), options => optionsBuilder((IFakeOptions<T>)options));
+            return (T)FakeAndDummyManager.CreateFake(new DummyCreationSession(), typeof(T), options => optionsBuilder((IFakeOptions<T>)options));

```

Comment:
Shouldn't it be injected at the `FakeAndDummyManager` level (i.e. one level "below")? Or maybe even at the `FakeObjectCreator` level? It seems a bit strange to me to create the `DummyCreationSession` at this level of abstraction; `A` shouldn't have to know that creating a fake (or dummy) requires a `DummyCreationSession`.

Also, moving it to a lower abstraction level means that less classes need to be modified (e.g. `A`, `Create`, and `Fake<T>` would be unaffected)